packages/react-tinacms-inline/src/ref-fields/use-field-ref.tsx

```diff
@@ -0,0 +1,32 @@
+/**
+
+Copyright 2019 Forestry.io Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+import * as React from 'react'
+import { useInlineForm } from 'inline-form'
+
+export type FieldRefType = React.RefObject<HTMLElement | null>
+
+export function useFieldRef(fieldName: string) {
+  const ref: FieldRefType = React.useRef(null)
+  const { fieldRefActions } = useInlineForm()
+  React.useEffect(() => {
+    fieldRefActions.set(fieldName, ref)
+  }, [fieldName, ref])
+
```

Comment:
Either that or we pass the `{onClick: () => setFocussedField(fieldName)}` back to the component along with the `ref`

Reply:
> Might feel a bit awkward to attach this to the raw onclick event, but I don't see anything wrong with it.

It _is_ awkward, but I feel it's worth doing to a limited degree because of how simple it makes things for the consumer.

Reply:
Yeah that's how I feel too, this ref-based approach in general is heavily based working with the actual DOM by its nature

Reply:
the solution as proposed isn't great because we should use `ref.current` in the dependency array since it's a mutable value and won't be subscribed correctly.

I've tried refactoring this to remove `FieldTarget` and tbh the resulting code is a lot messier and the styles aren't quite as clean.

Reply:
@jeffsee55 I made a separate PR with a set of changes to remove the need for `FieldTarget` as well as rendering a `FieldOverlay` for each field by listening to these native DOM events: https://github.com/tinacms/tinacms/pull/1741